recipe_scrapers/grandfrais.py

```diff
@@ -0,0 +1,102 @@
+# mypy: allow-untyped-defs
+
+
+import re
+from urllib.parse import urlparse
+
+from ._abstract import AbstractScraper
+from ._utils import FRACTIONS, get_minutes, normalize_string
+
+
+class GrandFrais(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "grandfrais.com"
+
+    def title(self):
+        return self.schema.title()
+
+    def category(self):
+        return None
+
+    def total_time(self):
+        cook_time, prep_time = self.cook_time(), self.prep_time()
+        if cook_time and prep_time:
+            return cook_time + prep_time
+        if prep_time:
+            return prep_time
+        return None
+
+    def cook_time(self):
+        raw = [
+            item
+            for item in self.soup.find_all("div", {"class": "pre-requie-item"})
+            if "Cuisson" in item.text
+        ]
+        if raw:
+            return get_minutes(raw[0].text)
+        return None
+
+    def prep_time(self):
+        raw = [
+            item
+            for item in self.soup.find_all("div", {"class": "pre-requie-item"})
+            if "Préparation" in item.text
+        ]
+        if raw:
+            return get_minutes(raw[0].text)
+        return None
+
+    def yields(self):
+        return self.schema.yields()
+
+    def image(self):
+        img = self.soup.find("img", {"alt": self.title()})
+        if img:
+            canonical = urlparse(self.canonical_url())
+            return f"{canonical.scheme}://{canonical.netloc}{img['src']}"
+        return None
+
+    def nutrients(self):
+        return None
+
+    def ingredients(self):
+        # Normalize all ingredients to have a decimal point, remove the prepended
+        # dash, and remove any trailing whitespace.
+        normalize = lambda x: re.sub(
+            r"^(\d+),(\d+)", r"\1.\2", normalize_string(x.lstrip("-").strip())
+        )
+
+        ingredients = [
+            normalize(i)
+            for i in self.soup.find("div", {"itemprop": "ingredients"}).text.split("\n")
+        ]
+
+        for fraction, replacement in FRACTIONS.items():
+            ingredients = [i.replace(fraction, str(replacement)) for i in ingredients]
```

Comment:
This seems OK; I'm wondering whether we should create a utility function that can do this.  But perhaps not until there are multiple places that do this kind of remapping.

Reply:
I was surprised it hadn't been used before for other websites.

Reply:
It might be interesting to try moving it into the `normalize_string` method, and then to run the unit tests to see how many sites are affected.

I don't think we've had any complaints about the library returning fractions from sites before, though, so I suppose people are (mostly?) OK with it so far.